extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusPermissionSecurityIdentityAugmentor.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.security.runtime;
+
+import java.security.Permission;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import io.quarkus.security.ForbiddenException;
+import io.quarkus.security.identity.AuthenticationRequestContext;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.SecurityIdentityAugmentor;
+import io.quarkus.security.spi.runtime.BlockingSecurityExecutor;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Adds a permission checker that grants access to the {@link QuarkusPermission}
+ * when {@link QuarkusPermission#isGranted(SecurityIdentity)} is true.
+ */
+final class QuarkusPermissionSecurityIdentityAugmentor implements SecurityIdentityAugmentor {
+
+    /**
+     * Permission checker only authorizes authenticated users and checkers shouldn't throw a security exception.
+     * However, it can happen than runtime exception occur, and we shouldn't leak that something wrong with response status.
+     */
+    private static final Predicate<Throwable> NOT_A_FORBIDDEN_EXCEPTION = new Predicate<>() {
+        @Override
+        public boolean test(Throwable throwable) {
+            return !(throwable instanceof ForbiddenException);
+        }
+    };
+    private static final Function<Throwable, Throwable> WRAP_WITH_FORBIDDEN_EXCEPTION = new Function<>() {
+        @Override
+        public Throwable apply(Throwable throwable) {
+            return new ForbiddenException(throwable);
+        }
+    };
+
+    private final BlockingSecurityExecutor blockingExecutor;
+
+    QuarkusPermissionSecurityIdentityAugmentor(BlockingSecurityExecutor blockingExecutor) {
+        this.blockingExecutor = blockingExecutor;
+    }
+
+    @Override
+    public Uni<SecurityIdentity> augment(SecurityIdentity identity, AuthenticationRequestContext context) {
+        if (identity.isAnonymous()) {
+            return Uni.createFrom().item(identity);
+        }
+
+        return Uni.createFrom().item(QuarkusSecurityIdentity
+                .builder(identity)
+                .addPermissionChecker(new Function<>() {
+                    @Override
+                    public Uni<Boolean> apply(Permission requiredpermission) {
+                        if (requiredpermission instanceof QuarkusPermission<?> quarkusPermission) {
+                            return quarkusPermission
+                                    .isGranted(identity, blockingExecutor)
```

Comment:
> I don't mind it given that it is an internal detail right now, but IMHO we should not proceed later with offering this option as an easier to understand augmented permission checker.

Sure, I think you were right there as checkers are fine to use. Maybe we can allow static checker methods on plain java classes, but only if there is ever demand from users.

Reply:
Also @sberyozkin , when I suggested it, I didn't know where I will iterate to, this looks differently than my original plan (that was simpler).

Reply:
@michalvavrik What I meant was that right now this solution is about the fact that augmenting the identity with permission checking functions is the only way to enforce permission based control, and here we do what looks nicer to the eye, but also block `implies`, the solution is constrained by the fact that the only way to do it is to use permission checker functions.
IMHO, adding the permission checkers should be avoidable even if no `@PermissionChecker` is used, as proposed in #43717. If #43717 is done next, then `QuarkusPermission`, `QuarkusPermissionSecurittyIdentityAugmentor` will probably become redundant, Quarkus security can just do 

```
for (Permission possessedPermission : identity.permissions()) {
possessedPermission.implies(requiredPermission);
}
```
somewhere next to the role based check.
Users will only see in the docs that they need to augment identity with possessed permissions, and not write correct permission checks, this is not what they should do...
Which is why IMHO we should keep `QuarkusPermission`, `QuarkusPermissionSecurittyIdentityAugmentor` low profile for now

Reply:
I agree with that and I did act upon that. Only `QuarkusPermission` is public but that cannot be avoided because I wanted to support declaring of permission checker methods on package-private CDI method beans. I think current state is ok and we should not proceed with my original proposal regarding `QuarkusPermission`.